README
Dist::Zilla::Role::BundleDeps

This role attempts to solve the problem of communicating dependencies to
META.* from bundles in a different way.

My first attempt was "[Prereqs::Plugins]", which added all values that
are seen in the "dist.ini" to dependencies.

However, that was inherently limited, as the ":version" specifier is
lost before the plugins appear on "$zilla->plugins"

This Role however, can see any declarations of ":version" your bundle
advertises, by standing between your "bundle_config" method and
"Dist::Zilla"

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-Role-BundleDeps

0.002006 2017-03-03T11:01:42Z 5b449df

0.002005 2017-03-03T10:25:23Z a10938d
 [Bugfix]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-Role-BundleDeps/issues/3

 [Dependencies::Stats]
 - Dependencies changed since 0.002004, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑1)
 - test: ↓1

0.002004 2016-05-20T23:18:11Z 7072374
 - Annotate documentation  to indicate need to add authordeps on bundles. ( PR #2, thanks to Olivier Mengué )

 [Dependencies::Stats]
 - Dependencies changed since 0.002003, see misc/*.deps* for details
 - configure: (suggests: +1)

0.002003 2016-05-01T03:40:22Z bbd3a71
 - No interesting changes.
 - Packaging cleaups, useless files etc, saves 4kb. If 4kb of savings is enough for Perl to break things, its good
   enough for me to at least justify a re-release.

 [Dependencies::Stats]
 - Dependencies changed since 0.002002, see misc/*.deps* for details
 - develop: +5 ↑1 -1 (suggests: ↑2)

0.002002 2015-05-05T22:26:50Z 2ee6cf4
 [Dependencies::Stats]
 - Dependencies changed since 0.002001, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +10 ↑3 -2 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1
 - test: +1 (recommends: +1 ↑1 ↓1)

 [Documentation]
 - Crossreference Dzil::Plugin::BundleInspector (Thanks Olivier Mengué/DOLMEN)
 - Closes #1

 [Metadata]
 - show 1x contributor
 - author = cpan

0.002001 2014-08-16T02:13:11Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - whitespace reduced.
 - metadata updated.
 - tests updated.

 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - configure: ↓1
 - develop: +1 (suggests: ↑1)
 - test: +1 -3 (recommends: ↑1 -1)

 [Metadata]
 - migrate to kentnl org.

 [Misc]
 - Improve source side POD to reduce build side whitespace.

 [Tests]
 - Updated Test::ReportPrereqs

0.002000 2014-07-14T13:02:42Z
 [00 Major]
 - Mostly changes that might affect downstreams that work by hand.

 [01 Tooling]
 - Switched to EUMM ( https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f )
 - Toolkit Dependencies softened to recommendations.

 [Dependencies::Stats]
 - Dependencies changed since 0.001001, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +9 -11 (suggests: +2 -1)
 - runtime: +1 ↑1
 - test: +4 ↓1 (recommends: +4)

0.001001 2013-11-24T15:32:47Z
 [00 Minor]
 - Maint release.

 [Dependencies::Noteworthy]
 - use Module::Build 0.4202
 - use Test::More 1.001002

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +1 ↑3 ↓1 -1
 - test: ↑1 -3

 [Documentation]
 - README.mkdn → metacpan links

 [Tests]
 - use Test::Compile::PerFile

0.001000 2013-11-04T18:04:56Z
 - First version.


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Automatically add all plugins in a bundle as dependencies",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Role-BundleDeps",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Role::BundleDeps",
  "PREREQ_PM" => {
    "CPAN::Meta::Requirements" => 0,
    "Moose::Role" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Role::PluginBundle" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Moose" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.002006",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::Meta::Requirements" => 0,
  "Dist::Zilla::Role::PluginBundle" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Moose" => 0,
  "Moose::Role" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');
#$bundle->remove_policy('CodeLayout::RequireUseUTF8');

#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.019.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4205"
  },
  "configure_requires" => {
    "Module::Build" => "0.4205"
  },
  "dist_abstract" => "Automatically add all plugins in a bundle as dependencies",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Role-BundleDeps",
  "dist_version" => "0.002000",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Role::BundleDeps",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "CPAN::Meta::Requirements" => 0,
    "Moose::Role" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Dist::Zilla::Role::PluginBundle" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Test::More" => "1.001003",
    "version" => 0
  }
);


my %fallback_build_requires = (
  "Dist::Zilla::Role::PluginBundle" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec::Functions" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.4205",
  "Moose" => 0,
  "Test::More" => "1.001003",
  "version" => 0
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
